src/overlays/actors/ovl_Bg_Lotus/z_bg_lotus.c

```diff
@@ -20,22 +22,154 @@ const ActorInit Bg_Lotus_InitVars = {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Lotus/func_80AD6A88.s")
+void BgLotus_Destroy(Actor* thisx, PlayState* play) {
+    BgLotus* this = THIS;
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Lotus/func_80AD6B68.s")
+void func_80AD6830(Actor* thisx) {
+    BgLotus* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Lotus/BgLotus_Update.s")
+    f32 temp_fv1;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Lotus/BgLotus_Draw.s")
+    if (thisx->params == 0) {
+        temp_fv1 = Math_SinF(this->unk166 * (M_PI / 4)) * ((0.014f * (this->unk166 / (80.0f * 0.1f))) + 0.01f);
+        thisx->scale.x = ((1.0f + temp_fv1) * 0.1f);
+        thisx->scale.z = ((1.0f - temp_fv1) * 0.1f);
+    }
+}
+
+void func_80AD68DC(BgLotus* thisx, PlayState* play) {
+    f32 sp34;
+
+    thisx->unk168 = (thisx->unk168 - 1);
+    sp34 = Math_SinF(thisx->unk168 * 0.06544985f) * 6.0f;
+    if (thisx->dyna.actor.params == 0) {
+        thisx->dyna.actor.world.pos.x =
+            (Math_SinS(thisx->dyna.actor.world.rot.y) * sp34) + thisx->dyna.actor.home.pos.x;
+        thisx->dyna.actor.world.pos.z =
+            (Math_CosS(thisx->dyna.actor.world.rot.y) * sp34) + thisx->dyna.actor.home.pos.z;
+        if (thisx->unk168 == 0) {
+            thisx->unk168 = 0x60;
+            thisx->dyna.actor.world.rot.y += (s16)((s32)Rand_Next() >> 0x12);
+        }
+    }
+    if (thisx->unk160 < thisx->dyna.actor.floorHeight) {
+        thisx->dyna.actor.world.pos.y = thisx->dyna.actor.floorHeight;
+        goto block_12;
+    }
+    thisx->dyna.actor.world.pos.y = thisx->unk160;
+    if (DynaPolyActor_IsPlayerOnTop((DynaPolyActor*)thisx) != 0) {
+        if (thisx->unk164 == 0) {
+            EffectSsGRipple_Spawn(play, &thisx->dyna.actor.world.pos, 0x3E8, 0x578, 0);
+            EffectSsGRipple_Spawn(play, &thisx->dyna.actor.world.pos, 0x3E8, 0x578, 8);
+            thisx->unk166 = 0x28;
+        }
+        if (gSaveContext.save.playerForm != 3) {
+            thisx->unk166 = 0x28;
+            thisx->dyna.actor.flags |= 0x10;
+            thisx->unk15C = func_80AD6A88;
+            return;
+        }
+        thisx->unk164 = 1U;
+        goto block_12;
+    }
+    thisx->unk164 = 0U;
+block_12:
+    if (thisx->unk166 > 0) {
+        thisx->unk166 = (thisx->unk166 - 1);
+    }
+    func_80AD6830(&thisx->dyna.actor);
+}
+
+void func_80AD6A88(Actor* thisx, PlayState* play) {
+    BgLotus* this = THIS;
+
+    if (this->unk160 < thisx->world.pos.y) {
+        thisx->world.pos.y = this->unk160;
+    }
+    thisx->world.pos.y -= 1.0f;
+    if (thisx->world.pos.y <= thisx->floorHeight) {
+        thisx->world.pos.y = thisx->floorHeight;
+        this->unk166 = 0;
+    }
+    if (this->unk166 > 0) {
+        this->unk166 = (this->unk166 - 1);
+        func_80AD6830(thisx);
+        return;
+    }
+    if (Math_StepToF(&thisx->scale.x, 0.0f, 0.0050000004f) != 0) {
```

Comment:
You might be able to shorten this, but might not work.

```suggestion
    if (Math_StepToF(&thisx->scale.x, 0.0f, 0.005f) != 0) {
```

Reply:
not working

Reply:
Alright no worries then